node/src/main/scala/com/wavesplatform/api/http/DebugApiRoute.scala

```
@@ -221,33 +220,32 @@ case class DebugApiRoute(
       val parsedTransaction = TransactionFactory.fromSignedRequest(jsv)
 
       val tracedDiff = for {
+        b <- TracedResult(blockchain)
         tx   <- TracedResult(parsedTransaction)
-        diff <- TransactionDiffer.forceValidate(blockchain.lastBlockTimestamp, time.correctedTime())(blockchain, tx)
-      } yield (tx, diff)
+        diff <- TransactionDiffer.forceValidate(b.lastBlockTimestamp, time.correctedTime())(b, tx)
+      } yield (tx, diff, b)
 
       val error = tracedDiff.resultE match {
-        case Right((tx, diff)) => diff.errorMessage(tx.id()).map(em => GenericError(em.text))
+        case Right((tx, diff, _)) => diff.errorMessage(tx.id()).map(em => GenericError(em.text))
         case Left(err)         => Some(err)
       }
 
       val transactionJson = parsedTransaction.fold(_ => jsv, _.json())
 
       val serializer = tracedDiff.resultE
-        .fold(_ => this.serializer, {
-          case (_, diff) =>
-            val compositeBlockchain = CompositeBlockchain(blockchain, diff)
-            this.serializer.copy(blockchain = compositeBlockchain)
-        })
+        .flatMap { case (_, diff, _) => blockchain.flatMap(CompositeBlockchain(_, diff)) }
+        .fold(_ => this.serializer, blockchain => this.serializer.copy(blockchain = blockchain))
```

This's interesting. Why can we ignore the fact that we aren't able to construct `CompositeBlockchain` (the first part of `fold`). This means we can't apply the `diff`, thus, the blockchain won't accept this transaction too.

It looks like we should return the error from this API in this case. Or add a comment, why this is not possible.

---

Not actual currently